test/e2e/page-objects/pages/app.ts

```diff
@@ -0,0 +1,33 @@
+import { Driver } from '../../webdriver/driver';
+import { WALLET_PASSWORD } from '../../helpers';
+import { BasePage } from './base-page';
+import { HeaderNavbar } from './header-navbar';
+import { SettingsPage } from './settings-page';
+import HomePage from './homepage';
+import LoginPage from './login-page';
+
+// Top level page object for the app.
+class App extends BasePage {
+  async getHeaderNavbar(): Promise<HeaderNavbar> {
+    // TODO: Ensure that the header navbar is visible.
+    return new HeaderNavbar(this.driver);
+  }
+
+  async getLoginPage(): Promise<LoginPage> {
+    return new LoginPage(this.driver).check_pageIsLoaded();
+  }
+
+  async openSettings(): Promise<SettingsPage> {
+    const headerNavbar = await this.getHeaderNavbar();
+    return headerNavbar.openSettings();
+  }
+
+  async login(password: string = WALLET_PASSWORD): Promise<HomePage> {
+    const loginPage = await this.getLoginPage();
+    return loginPage.login(password);
+  }
+}
+
+export const getApp = (driver: Driver): App => {
+  return new App(driver);
+};
```

Comment:
I'm wondering if we really need to add this class. For the `getHeaderNavbar()` and `getLoginPage()` methods, we can create an instance of` HeaderNavbar()` or `LoginPage()` directly when we need them. For the `openSettings()`, we can do `new HeaderNavbar(this.driver).openSettings()` in tests. For the login method, I think the original `loginWithBalanceValidation` is more straightforward and efficient enough.
As page object model is in the adoption phase, we are trying to keep it as simple as possible to facilitate easier onboarding and usage.